internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangeLogConsumerImpl.java

```diff
@@ -0,0 +1,288 @@
+    CompletableFuture<Void> completableFuture = CompletableFuture.supplyAsync(() -> {
+      Set<TopicPartition> topicPartitionSet = kafkaConsumer.assignment();
+      List<TopicPartition> topicPartitionList = getPartitionListToSubscribe(partitions, topicPartitionSet);
+      kafkaConsumer.assign(topicPartitionList);
+      kafkaConsumer.seekToBeginning(topicPartitionList);
+      subscribedPartitions.addAll(partitions);
+      return null;
+    });
+    return completableFuture;
+  }
+
+  @Override
+  public CompletableFuture<Void> subscribeAll() {
+    Set<Integer> allPartitions = new HashSet<>();
+    for (int partition = 0; partition < partitionCount; partition++) {
+      allPartitions.add(partition);
+    }
+    return this.subscribe(allPartitions);
+  }
+
+  private List<TopicPartition> getPartitionListToSubscribe(
+      Set<Integer> partitions,
+      Set<TopicPartition> topicPartitionSet) {
+    List<TopicPartition> topicPartitionList = new ArrayList<>(topicPartitionSet);
+    for (Integer partition: partitions) {
+      TopicPartition topicPartition = new TopicPartition(currentTopic, partition);
+      if (!topicPartitionSet.contains(topicPartition)) {
+        topicPartitionList.add(topicPartition);
+      }
+    }
+    return topicPartitionList;
+  }
+
+  @Override
+  public void unsubscribe(Set<Integer> partitions) {
+    for (Integer partition: partitions) {
+      TopicPartition topicPartition = new TopicPartition(currentTopic, partition);
+      Set<TopicPartition> topicPartitionSet = kafkaConsumer.assignment();
+      if (topicPartitionSet.contains(topicPartition)) {
+        List<TopicPartition> topicPartitionList = new ArrayList<>(topicPartitionSet);
+        if (topicPartitionList.remove(topicPartition)) {
+          kafkaConsumer.assign(topicPartitionList);
+        }
+      }
+      subscribedPartitions.remove(topicPartition.partition());
+    }
+  }
+
+  @Override
+  public void unsubscribeAll() {
+    Set<Integer> allPartitions = new HashSet<>();
+    for (int partition = 0; partition < partitionCount; partition++) {
+      allPartitions.add(partition);
+    }
+    this.unsubscribe(allPartitions);
+  }
+
+  @Override
+  public Collection<PubSubMessage> poll(long timeoutInMs) {
+    Collection<PubSubMessage> pubSubMessages = new ArrayList<>();
+    ConsumerRecords<KafkaKey, KafkaMessageEnvelope> consumerRecords = kafkaConsumer.poll(timeoutInMs);
+    for (ConsumerRecord<KafkaKey, KafkaMessageEnvelope> consumerRecord: consumerRecords) {
+      PubSubTopicPartition pubSubTopicPartition = getPubSubTopicPartitionFromConsumerRecord(consumerRecord);
+      Long offset = consumerRecord.offset();
+      int payloadSize = consumerRecord.serializedKeySize() + consumerRecord.serializedValueSize();
+      long timestamp = consumerRecord.timestamp();
+      if (viewClassName.equals(ChangeCaptureView.class.getCanonicalName())) {
+        if (consumerRecord.key().isControlMessage()) {
```

Comment:
Lets move this out of the if/else for the view type to avoid repeating it.  We should honor the controlMessage regardless of the view type.